### DISCUSSION TITLE
removing missing values from coloumns

### DISCUSSION THREAD

**Discussion by Hasham04** [vaexio/vaex#1919] [Q&A] [answered]
col1               col2
value1             ''
''                     value1
value2              ''
''                         value2

any suggestions on how i can i convert this data to 

col1     col2
value1 value1
value2 value2

I then want to be able to export it to a CSV.

thankyou

**Comment by Hasham04** [comment, marked as answer]
One approach that I tried was to filter each column so 

df_col1_filtered=df_unfiltered[df_unfiltered.col1!='']
df_col2_filtered=df_unfiltered[df_unfiltered.col2!='']


when I describe this data I get an equal number of counts of rows. but when I try to join them like this.

df_col1_filtered.drop(col2,inplace=True)
df_col2_filtered.drop(col1,inplace=True)

joined=df_col1_filtered.join(df_col2_filtered)

I get an unequal number of rows being merged error. 

I would appreciate any suggestions to solve the problem I mentioned in question or any insight on what I am doing wrong with my approach. Thank you for your help.

**Reply by maartenbreddels** [reply]
your approach seems right! 👍 
if you can give us a reproducible example, we might be able to tell you more.

**Reply by JovanVeljanoski** [reply]
Perhaps you need to do `.extract()` after filtering the dataframes?

**Reply by Hasham04** [reply]
You guys are amazing. So helpful. thankyou.

Jovans Suggestion worked I had to extract the filters. Previously to make this work and to get around the an unequal number of rows being merged error I was converting both filtered dataframes to pandas and converting them vaex and then joining them which worked too hahah.

**Reply by Hasham04** [reply]
im using df.func.where(condition,val1,val2) to make the columns i mentioned above.

how do I pass na to val2 so that i can use dropna.

**Reply by maartenbreddels** [reply]
you can pass `None`

**Reply by Hasham04** [reply]
i have already tried that but i get 

None is not a an Arrow or NumPy array

**Reply by JovanVeljanoski** [reply]
This is answered somewhere on this issue board. But why do you wanna pass NA to the output of a `where` statement?

**Reply by Hasham04** [reply]
i have data that looks like this

col1 col2 col3 col4
val1 val2  val3 val4

making this data look like this 

with the where statement
df[newcol1]=df.func.where(col1=='something',col2,'')
df[newcol2]=df.func.where(col3=='something',col4,'')

col1 col2 col3 col4 newcol1 newcol2

I then drop  col1 col2 col3 col4.

which makes the data look like this.
newcol1              newcol2
value1                      ''
''                        value1
value2                      ''
''                          value2.

i am then filtering each column right now like this

df.newcol2=df_unfiltered[df.newcol2!='']

Ii was wondering if could do

newcol1=newcol1.dropna()
so I was wondering instead of filtering out '' if I could just use dropna .
